### PR TITLE
Describe the art of using git

### LINKED ISSUE #3: Describe commit messages
Describe how to structure and message a commit.

Acceptable commit messages:
 - Fix appointment picker bugs.

Bad commit messages:
 - Apply review comments.
 - A minor issue.
 - Some changes.
 - Solve a bug related to ...

### PR DESCRIPTION
**Pull request by stocarul**
- Describe the commit content and commit message best practices
 - Present few good and bad examples for commit messages
 - Closes #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,23 @@
...
+ - Commit message
+   - Should clearly specify the essence of the commit
+   - Describe the change (what the code does), not what you have done
+   - Every commit is a sentence in the repository history
+   - Always start with a capital letter and finish with a period
+   - Use the imperative present tense (fix, add, implement) describing the change in a small subject sentence
+   - Good examples:
+     1. Send an email to the user with a link to renew the password.
+     2. Add the newsletter signup in homepage.
+   - Bad examples:
```
Comment (lgnstn) [inline README.md]: These "bad examples" don't look that bad to me. Both of them make sense and can describe the work one has done.

Comment (stocarul) [reply README.md]: I completely agree with you @lgnstn. Actually, those bad examples are way better than some of the commits I see during code reviews. The problem with them is that they describe what the developer has done and not what the code does. Anyway, I think we should discuss this in a meeting. Maybe it is too difficult to achieve this.

File: README.md [hunk 2]
```diff
@@ -1 +1,23 @@
...
+
+### Git
+
+#### The art of committing
+
+ - Commit content
+   - A logically connected change
+   - Must not fail a build
+   - Break up every work into small tasks (no more than few hours of work per commit)
+   - Isolate meaningless commits
```
Comment (calina-c) [inline README.md]: I think we should clarify what "meaningless" is.

File: README.md (hunk 1 above)
Comment (stocarul) [reply README.md]: Maybe we should give examples for good, acceptable and bad commits.

File: README.md [hunk 3]
```diff
@@ -1 +1,23 @@
...
+ - Commit message
```
Comment (OroianRares) [inline README.md]: Although this is somehow dependent on the size of the commit, I think we should say something about the commit message maximum length too.

**Pull request merged**